Lib/test/test_ctypes/test_bitfields.py

```diff
@@ -42,18 +67,43 @@ def test_ints(self):
                 setattr(b, name, i)
                 self.assertEqual(getattr(b, name), func(byref(b), name.encode('ascii')))
 
-    # bpo-46913: _ctypes/cfield.c h_get() has an undefined behavior
-    @support.skip_if_sanitizer(ub=True)
     def test_shorts(self):
         b = BITS()
         name = "M"
+        # See Modules/_ctypes/_ctypes_test.c for where the magic 999 comes from.
         if func(byref(b), name.encode('ascii')) == 999:
+            # unpack_bitfields and unpack_bitfields_msvc in
+            # Modules/_ctypes/_ctypes_test.c return 999 to indicate
+            # an invalid name. 'M' is only valid, if signed short bitfields
+            # are supported by the C compiler.
             self.skipTest("Compiler does not support signed short bitfields")
         for i in range(256):
             for name in "MNOPQRS":
                 b = BITS()
                 setattr(b, name, i)
-                self.assertEqual(getattr(b, name), func(byref(b), name.encode('ascii')))
+                self.assertEqual(
+                    getattr(b, name),
+                    func(byref(b), (name.encode('ascii'))),
+                    (name, i))
```

Comment:
consider updating tests like this that loop over byte values into sub-tests via the https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests API call so that you don't need to pass `msg=(name, i)` here to get a meaningful error message.